diff -wr -U 4 a/source/tools/toolutil/flagparser.c b/source/tools/toolutil/flagparser.c
--- a/source/tools/toolutil/flagparser.c	2015-10-08 06:54:06.000000000 +0300
+++ b/source/tools/toolutil/flagparser.c	2015-12-26 00:04:09.000000000 +0300
@@ -96,9 +96,9 @@
     uprv_free(buffer);

     T_FileStream_close(f);

-    if (U_FAILURE(*status)) {
+    if (U_FAILURE(*status) && (*status != U_BUFFER_OVERFLOW_ERROR)) {
         return -1;
     }

     if (U_SUCCESS(*status) && result == 0) {
