diff -r -U 3 a/Makefile.in b/Makefile.in
--- a/Makefile.in	2016-06-16 16:51:02.000000000 +0300
+++ b/Makefile.in	2016-06-16 16:48:57.000000000 +0300
@@ -14388,7 +14388,7 @@
 	$(AM_V_GEN)name=`echo $@ | sed 's|.*/||; s|\.1$$||'` || exit 1;	\
 	case $$name in							\
 	  install) prog='ginstall'; argv=$$name;;			\
-	     test) prog='['; argv='[';;					\
+		 test) prog='['; argv='[';;					\
 		*) prog=$$name; argv=$$prog;;				\
 	esac;								\
 	rm -f $@ $@-t							\
@@ -14399,15 +14399,15 @@
 				$$argv$(EXEEXT))			\
 	&& : $${SOURCE_DATE_EPOCH=`cat $(srcdir)/.timestamp 2>/dev/null || :`} \
 	&& export SOURCE_DATE_EPOCH && $(run_help2man)			\
-		     --source='$(PACKAGE_STRING)'			\
-		     --include=$(srcdir)/man/$$name.x			\
-		     --output=$$t/$$name.1				\
-		     --info-page='\(aq(coreutils) '$$name' invocation\(aq' \
-		     $$t/$$argv$(EXEEXT)				\
+			 --source='$(PACKAGE_STRING)'			\
+			 --include=$(srcdir)/man/$$name.x			\
+			 --output=$$t/$$name.1				\
+			 --info-page='\(aq(coreutils) '$$name' invocation\(aq' \
+			 $$t/$$argv$(EXEEXT)				\
 	  && sed \
-	       -e 's|$*\.td/||g' \
-	       -e '/For complete documentation/d' \
-	       $$t/$$name.1 > $@-t			\
+		   -e 's|$*\.td/||g' \
+		   -e '/For complete documentation/d' \
+		   $$t/$$name.1 > $@-t			\
 	  && rm -rf $$t							\
 	  && chmod a-w $@-t						\
 	  && mv $@-t $@
diff -r -U 3 a/configure b/configure
--- a/configure	2016-06-16 16:51:02.000000000 +0300
+++ b/configure	2016-06-16 16:48:57.000000000 +0300
@@ -34403,7 +34403,8 @@
        mint*)           os='MiNT';;
        mingw*)          os='MinGW';;
        lynxos*)         os='LynxOS';;
-       linux*)          os='GNU/Linux';;
+       linux*-gnu*)     os='GNU/Linux';;
+       linux*-android*) os='Android';;
        hpux*)           os='HP-UX';;
        hiux*)           os='HI-UX';;
        gnu*)            os='GNU';;
diff -r -U 3 a/lib/stdio-impl.h b/lib/stdio-impl.h
--- a/lib/stdio-impl.h	2016-06-16 16:51:02.000000000 +0300
+++ b/lib/stdio-impl.h	2016-06-16 16:48:58.000000000 +0300
@@ -59,6 +59,9 @@
 # endif
 
 # if (defined __NetBSD__ && __NetBSD_Version__ >= 105270000) || defined __OpenBSD__ || defined __minix || defined __ANDROID__ /* NetBSD >= 1.5ZA, OpenBSD, Minix 3, Android */
+#  if defined(__ANDROID__) && defined(__CRYSTAX__)
+#  define fp_ub fp_->_ub
+#  else
   /* See <http://cvsweb.netbsd.org/bsdweb.cgi/src/lib/libc/stdio/fileext.h?rev=HEAD&content-type=text/x-cvsweb-markup>
      and <http://www.openbsd.org/cgi-bin/cvsweb/src/lib/libc/stdio/fileext.h?rev=HEAD&content-type=text/x-cvsweb-markup> */
   struct __sfileext
@@ -67,6 +70,7 @@
       /* More fields, not relevant here.  */
     };
 #  define fp_ub ((struct __sfileext *) fp->_ext._base)->_ub
+#  endif
 # else                                         /* FreeBSD, NetBSD <= 1.5Z, DragonFly, Mac OS X, Cygwin, Android */
 #  define fp_ub fp_->_ub
 # endif
diff -r -U 3 a/m4/host-os.m4 b/m4/host-os.m4
--- a/m4/host-os.m4	2016-06-16 16:51:02.000000000 +0300
+++ b/m4/host-os.m4	2016-06-16 16:48:58.000000000 +0300
@@ -48,7 +48,8 @@
        mint*)           os='MiNT';;
        mingw*)          os='MinGW';;
        lynxos*)         os='LynxOS';;
-       linux*)          os='GNU/Linux';;
+       linux*-gnu*)     os='GNU/Linux';;
+       linux*-android*) os='Android';;
        hpux*)           os='HP-UX';;
        hiux*)           os='HI-UX';;
        gnu*)            os='GNU';;
diff -r -U 3 a/man/help2man b/man/help2man
--- a/man/help2man	2016-06-16 16:51:02.000000000 +0300
+++ b/man/help2man	2016-06-16 16:48:58.000000000 +0300
@@ -742,22 +742,24 @@
 # Call program with given option and return results.
 sub get_option_value
 {
-    my ($prog, $opt) = @_;
-    my $stderr = $discard_stderr ? '/dev/null' : '&1';
-    my $value = join '',
-	map { s/ +$//; expand $_ }
-	map { dec $_ }
-	`$prog $opt 2>$stderr`;
+    #my ($prog, $opt) = @_;
+    #my $stderr = $discard_stderr ? '/dev/null' : '&1';
+    #my $value = join '',
+	#map { s/ +$//; expand $_ }
+	#map { dec $_ }
+	#`$prog $opt 2>$stderr`;
 
-    unless ($value)
-    {
-	my $err = N_("%s: can't get `%s' info from %s%s");
-	my $extra = $discard_stderr
-	    ? "\n" . N_("Try `--no-discard-stderr' if option outputs to stderr")
-	    : '';
+    #unless ($value)
+    #{
+	#my $err = N_("%s: can't get `%s' info from %s%s");
+	#my $extra = $discard_stderr
+	#    ? "\n" . N_("Try `--no-discard-stderr' if option outputs to stderr")
+	#    : '';
 
-	kark $err, $this_program, $opt, $prog, $extra;
-    }
+	#kark $err, $this_program, $opt, $prog, $extra;
+    #}
+
+    my $value = '';
 
     $value;
 }
diff -r -U 3 a/src/nohup.c b/src/nohup.c
--- a/src/nohup.c	2016-06-16 16:51:02.000000000 +0300
+++ b/src/nohup.c	2016-06-16 16:48:58.000000000 +0300
@@ -138,7 +138,7 @@
       char const *file = "nohup.out";
       int flags = O_CREAT | O_WRONLY | O_APPEND;
       mode_t mode = S_IRUSR | S_IWUSR;
-      mode_t umask_value = umask (~mode);
+      mode_t umask_value = umask (~mode & 0777);
       out_fd = (redirecting_stdout
                 ? fd_reopen (STDOUT_FILENO, file, flags, mode)
                 : open (file, flags, mode));
