diff -Nabur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2018-05-30 16:47:04.781290279 +0300
+++ b/CMakeLists.txt	2018-05-30 17:18:24.569467731 +0300
@@ -1,4 +1,4 @@
-# Copyright (C) 2009, 2016 Julian Andres Klode <jak@debian.org>.
+# Copyright (C) 2009, 2016, 2018 Julian Andres Klode <jak@debian.org>.
 # Licensed under the same terms as APT; i.e. GPL 2 or later.
 
 # set minimum version
@@ -33,10 +33,10 @@
 include(GNUInstallDirs)
 include(TestBigEndian)
 find_package(Threads REQUIRED)
-find_package(LFS REQUIRED)
+#find_package(LFS REQUIRED)
 find_package(Iconv REQUIRED)
 
-find_package(Perl REQUIRED)
+#find_package(Perl REQUIRED)
 
 if(USE_NLS)
   find_package(Intl REQUIRED)
@@ -226,11 +226,11 @@
 add_subdirectory(apt-inst)
 add_subdirectory(cmdline)
 add_subdirectory(completions)
-add_subdirectory(doc)
+#add_subdirectory(doc)
 add_subdirectory(dselect)
-add_subdirectory(ftparchive)
+#add_subdirectory(ftparchive)
 add_subdirectory(methods)
-add_subdirectory(test)
+#add_subdirectory(test)
 
 if (USE_NLS)
 add_subdirectory(po)
