--- a/apt-pkg/CMakeLists.txt	2017-10-26 18:38:39.000000000 +0700
+++ b/apt-pkg/CMakeLists.txt	2018-02-25 15:57:51.704470795 +0700
@@ -49,7 +49,7 @@
 )
 
 target_link_libraries(apt-pkg
-                      PRIVATE -lutil ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
+                      PRIVATE ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
                              ${CMAKE_THREAD_LIBS_INIT}
                              ${ZLIB_LIBRARIES}
                              ${BZIP2_LIBRARIES}
