diff -Naur a/ext/extmk.rb b/ext/extmk.rb
--- a/ext/extmk.rb	2018-01-11 18:31:29.005791867 +0300
+++ b/ext/extmk.rb	2018-01-11 18:46:23.939488128 +0300
@@ -638,7 +638,7 @@
 FileUtils.makedirs(File.dirname($command_output))
 begin
   atomic_write_open($command_output) do |mf|
-    mf.puts "V = 0"
+    mf.puts "V = 1"
     mf.puts "Q1 = $(V:1=)"
     mf.puts "Q = $(Q1:0=@)"
     mf.puts "ECHO1 = $(V:1=@:)"
