From 8943f08cbb70fb5a02dbf98e47bd993e55ddf6fe Mon Sep 17 00:00:00 2001
From: Alexander Zhukov <zuav@crystax.net>
Date: Mon, 1 Jun 2015 17:32:45 +0300
Subject: [PATCH 1/4] Add makefile to build libgit2

---
 Makefile.crystax | 47 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 47 insertions(+)
 create mode 100644 Makefile.crystax

diff --git a/Makefile.crystax b/Makefile.crystax
new file mode 100644
index 0000000..4bce6e5
--- /dev/null
+++ b/Makefile.crystax
@@ -0,0 +1,47 @@
+# -*- makefile-gmake -*-
+
+ifeq ($(CC),clang)
+PREFIX =
+else
+PREFIX = $(subst -gcc,-,$(CC))
+endif
+
+$(info CC = $(CC))
+$(info PREFIX = $(PREFIX))
+
+RM = rm -f
+AR = $(PREFIX)ar cq
+RANLIB = $(PREFIX)ranlib
+
+LIBNAME=libgit2.a
+
+INCLUDES= -I. -Isrc -Iinclude -Ideps/http-parser -Ideps/zlib $(EXTRA_INCLUDES)
+DEFINES= $(INCLUDES) -DNO_VIZ -DSTDC -DNO_GZIP -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE $(EXTRA_DEFINES)
+CFLAGS= -g $(DEFINES) -Wall -Wextra -Wno-missing-field-initializers -O2 $(EXTRA_CFLAGS)
+
+#  src/hash/hash_generic.c
+SRCS = $(wildcard src/*.c) $(wildcard src/transports/*.c) $(wildcard src/xdiff/*.c) $(wildcard deps/http-parser/*.c) $(wildcard deps/zlib/*.c)
+
+ifeq ($(MINGW),1)
+  SRCS += $(wildcard src/win32/*.c) $(wildcard src/compat/*.c) deps/regex/regex.c
+  INCLUDES += -Ideps/regex
+  DEFINES += -DWIN32 -D_WIN32_WINNT=0x0501 -D__USE_MINGW_ANSI_STDIO=1
+else
+  SRCS += $(wildcard src/unix/*.c)
+  #CFLAGS += -fPIC
+endif
+
+OBJS = $(patsubst %.c,%.o,$(SRCS))
+
+%.c.o:
+	$(CC) $(CFLAGS) -c $*.c
+
+all: $(LIBNAME)
+
+$(LIBNAME): $(OBJS)
+	$(RM) $@
+	$(AR) $@ $(OBJS)
+	$(RANLIB) $@
+
+clean:
+	$(RM) $(OBJS) $(LIBNAME)
-- 
2.7.0

