From 68f77db3f7ed4d67ccd70ae28f0021c8d2da2205 Mon Sep 17 00:00:00 2001
From: Dmitry Moskalchuk <dm@crystax.net>
Date: Mon, 13 Apr 2015 19:14:33 +0300
Subject: [PATCH] Undef MIN/MAX macros to be able build without warnings

Signed-off-by: Dmitry Moskalchuk <dm@crystax.net>
---
 sqlite3.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sqlite3.c b/sqlite3.c
index a8922e4..84b9c66 100644
--- a/sqlite3.c
+++ b/sqlite3.c
@@ -8811,7 +8811,14 @@ SQLITE_PRIVATE void sqlite3HashClear(Hash*);
 /*
 ** Macros to compute minimum and maximum of two numbers.
 */
+#ifdef MIN
+#undef MIN
+#endif
 #define MIN(A,B) ((A)<(B)?(A):(B))
+
+#ifdef MAX
+#undef MAX
+#endif
 #define MAX(A,B) ((A)>(B)?(A):(B))
 
 /*
-- 
2.7.0

