diff -Naurw a/g10/t-stutter.c b/g10/t-stutter.c
--- a/g10/t-stutter.c	2017-08-28 13:22:54.000000000 +0300
+++ b/g10/t-stutter.c	2019-03-25 16:47:25.054790049 +0300
@@ -195,7 +195,7 @@
 /* Whether we need to resynchronize the CFB after writing the random
    data (this is the case for encrypted packets, but not encrypted and
    integrity protected packets).  */
-static int sync;
+static int cfb_sync;
 
 static int
 block_offset (int i)
@@ -255,7 +255,7 @@
 
   rc = openpgp_cipher_open (&cipher_hd, dek.algo,
 			    GCRY_CIPHER_MODE_CFB,
-			    (! sync /* ed->mdc_method || dek.algo >= 100 */ ?
+			    (! cfb_sync /* ed->mdc_method || dek.algo >= 100 */ ?
                              0 : GCRY_CIPHER_ENABLE_SYNC));
   if (rc)
     log_fatal ("Failed to open cipher: %s\n", gpg_strerror (rc));
@@ -485,10 +485,10 @@
            is 1.  */
         log_assert (raw_data[hdrlen] == 1);
         hdrlen ++;
-        sync = 0;
+        cfb_sync = 0;
       }
     else
-      sync = 1;
+      cfb_sync = 1;
 
     msg = &raw_data[hdrlen];
     msg_len = raw_data_len - hdrlen;
