diff -Naurw a/configure b/configure
--- a/configure	2018-07-26 19:58:08.381656619 +0300
+++ b/configure	2018-07-26 19:58:31.510236058 +0300
@@ -16569,12 +16569,12 @@
 
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether issetugid() can detect setuid/setgid" >&5
 $as_echo_n "checking whether issetugid() can detect setuid/setgid... " >&6; }
-		issetugid_openbsd=no
+if ${ac_cv_issetugid_openbsd+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_cv_issetugid_openbsd=no
 		if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  ac_cv_issetugid_openbsd="guessing no"
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -16588,19 +16588,22 @@
 }
 _ACEOF
 if ac_fn_c_try_run "$LINENO"; then :
-  chmod 02777 ./conftest$EXEEXT; ./conftest$EXEEXT || issetugid_openbsd=yes
+  chmod 02777 ./conftest$EXEEXT; ./conftest$EXEEXT || ac_cv_issetugid_openbsd=yes
+else
+  ac_cv_issetugid_openbsd=no
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
 
-		if test "$issetugid_openbsd" = yes; then
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_issetugid_openbsd" >&5
+$as_echo "$ac_cv_issetugid_openbsd" >&6; }
+		if test "$ac_cv_issetugid_openbsd" = yes; then
 
 $as_echo "#define HAVE_ISSETUGID_OPENBSD 1" >>confdefs.h
 
 		fi
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $issetugid_openbsd" >&5
-$as_echo "$issetugid_openbsd" >&6; }
 
 fi
 done
