diff -Naurw a/bzip2.c b/bzip2.c
--- a/bzip2.c	2010-09-11 03:04:53.000000000 +0400
+++ b/bzip2.c	2019-06-12 16:29:14.693930752 +0300
@@ -34,7 +34,7 @@
 --*/
 #define BZ_LCCWIN32  0
 
-#if defined(_WIN32) && !defined(__CYGWIN__)
+#if defined(_WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #undef  BZ_LCCWIN32
 #define BZ_LCCWIN32 1
 #undef  BZ_UNIX
@@ -72,10 +72,17 @@
 #   include <utime.h>
 #   include <unistd.h>
 #   include <sys/stat.h>
+
+#   ifndef __MINGW32__
 #   include <sys/times.h>
+#   endif
 
 #   define PATH_SEP    '/'
+#   ifndef __MINGW32__
 #   define MY_LSTAT    lstat
+#   else
+#     define MY_LSTAT    stat
+#endif
 #   define MY_STAT     stat
 #   define MY_S_ISREG  S_ISREG
 #   define MY_S_ISDIR  S_ISDIR
@@ -1065,6 +1072,7 @@
 void applySavedFileAttrToOutputFile ( IntNative fd )
 {
 #  if BZ_UNIX
+#  ifndef __MINGW32__
    IntNative retVal;
 
    retVal = fchmod ( fd, fileMetaInfo.st_mode );
@@ -1075,6 +1083,7 @@
       be safely ignored.
    */
 #  endif
+#  endif
 }
 
 
@@ -1808,9 +1817,11 @@
    signal (SIGSEGV, mySIGSEGVorSIGBUScatcher);
 #  if BZ_UNIX
 #  ifndef __DJGPP__
+#  ifndef __MINGW32__
    signal (SIGBUS,  mySIGSEGVorSIGBUScatcher);
 #  endif
 #  endif
+#  endif
 
    copyFileName ( inName,  (Char*)"(none)" );
    copyFileName ( outName, (Char*)"(none)" );
@@ -1952,8 +1963,10 @@
       signal (SIGINT,  mySignalCatcher);
       signal (SIGTERM, mySignalCatcher);
 #     if BZ_UNIX
+#     ifndef __MINGW32__
       signal (SIGHUP,  mySignalCatcher);
 #     endif
+#     endif
    }
 
    if (opMode == OM_Z) {
