diff -Naur a/libarchive/archive_cryptor_private.h b/libarchive/archive_cryptor_private.h
--- a/libarchive/archive_cryptor_private.h	2017-01-29 08:06:27.000000000 +0300
+++ b/libarchive/archive_cryptor_private.h	2018-01-06 17:51:38.412184061 +0300
@@ -1,5 +1,5 @@
 /*-
-* Copyright (c) 2014 Michihiro NAKAJIMA
+* Copyright (c) 2014, 2018 Michihiro NAKAJIMA
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
@@ -64,7 +64,7 @@
 } archive_crypto_ctx;
 
 #elif defined(_WIN32) && !defined(__CYGWIN__) && defined(HAVE_BCRYPT_H)
-#include <Bcrypt.h>
+#include <bcrypt.h>
 
 /* Common in other bcrypt implementations, but missing from VS2008. */
 #ifndef BCRYPT_SUCCESS
