diff -Naur a/libs/python/build/Jamfile b/libs/python/build/Jamfile
--- a/libs/python/build/Jamfile	2016-12-22 15:33:19.000000000 +0300
+++ b/libs/python/build/Jamfile	2017-04-20 14:24:25.000000000 +0300
@@ -1,4 +1,4 @@
-# Copyright David Abrahams 2001-2006. Distributed under the Boost
+# Copyright David Abrahams 2001-2006, 2017. Distributed under the Boost
 # Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
@@ -162,14 +162,17 @@
     libraries3 += boost_numpy3 ;
 }
 
-lib_boost_python ;
-lib_boost_numpy ;
 
 if $(py3-version)
 {
     lib_boost_python yes ;
     lib_boost_numpy yes ;
-    libraries += $(libraries3) ;
+    libraries = $(libraries3) ;
+}
+else
+{
+    lib_boost_python ;
+    lib_boost_numpy ;
 }
 
 boost-install $(libraries) ;
