diff -Naur a/configure b/configure
--- a/configure	2017-12-03 20:43:24.000000000 +0300
+++ b/configure	2018-05-11 20:47:26.562009973 +0300
@@ -5,7 +5,7 @@
 # Report bugs to <https://bugs.gnupg.org>.
 #
 #
-# Copyright (C) 1992-1996, 1998-2012 Free Software Foundation, Inc.
+# Copyright (C) 1992-1996, 1998-2012, 2018 Free Software Foundation, Inc.
 #
 #
 # This configure script is free software; the Free Software Foundation
@@ -6989,6 +6989,11 @@
 NEED_GPG_ERROR_VERSION=1.16
 have_gpg_error=no
 
+if test -n "$GPG_ERROR_CFLAGS"; then
+    $as_echo "GPG_ERROR_CFLAGS: $GPG_ERROR_CFLAGS"
+    $as_echo "GPG_ERROR_LIBS:   $GPG_ERROR_LIBS"
+    have_gpg_error=yes
+else
 
   gpg_error_config_prefix=""
 
@@ -7131,7 +7136,7 @@
     have_gpg_error=no
   fi
 
-
+fi
 
 
 
@@ -7158,6 +7163,12 @@
 NEED_LIBASSUAN_VERSION=2.1.0
 have_libassuan=no
 
+if test -n "$LIBASSUAN_CFLAGS"; then
+    $as_echo "LIBASSUAN_CFLAGS: $LIBASSUAN_CFLAGS"
+    $as_echo "LIBASSUAN_LIBS:   $LIBASSUAN_LIBS"
+    have_libassuan=yes
+    libassuan_version=$LIBASSUAN_VERSION
+else
 
 # Check whether --with-libassuan-prefix was given.
 if test "${with_libassuan_prefix+set}" = set; then :
@@ -7394,6 +7405,7 @@
     have_libassuan=no
   fi
 
+fi
 
 
 if test "$have_libassuan" = "yes"; then
