diff -Naurw a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2018-04-20 13:08:18.000000000 +0300
+++ b/CMakeLists.txt	2018-06-14 16:24:13.390708313 +0300
@@ -1,4 +1,4 @@
-# Copyright (C) 2009, 2016 Julian Andres Klode <jak@debian.org>.
+# Copyright (C) 2009, 2016, 2018 Julian Andres Klode <jak@debian.org>.
 # Licensed under the same terms as APT; i.e. GPL 2 or later.
 
 # set minimum version
@@ -33,11 +33,8 @@
 include(GNUInstallDirs)
 include(TestBigEndian)
 find_package(Threads REQUIRED)
-find_package(LFS REQUIRED)
 find_package(Iconv REQUIRED)
 
-find_package(Perl REQUIRED)
-
 if(USE_NLS)
   find_package(Intl REQUIRED)
   link_libraries(${Intl_LIBRARIES})
@@ -221,11 +218,8 @@
 add_subdirectory(apt-inst)
 add_subdirectory(cmdline)
 add_subdirectory(completions)
-add_subdirectory(doc)
 add_subdirectory(dselect)
-add_subdirectory(ftparchive)
 add_subdirectory(methods)
-add_subdirectory(test)
 
 if (USE_NLS)
 add_subdirectory(po)
