diff -Naurw a/apt-pkg/CMakeLists.txt b/apt-pkg/CMakeLists.txt
--- a/apt-pkg/CMakeLists.txt	2018-04-20 13:08:18.000000000 +0300
+++ b/apt-pkg/CMakeLists.txt	2018-06-14 16:28:07.306234484 +0300
@@ -50,7 +50,7 @@
 )
 
 target_link_libraries(apt-pkg
-                      PRIVATE -lutil ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
+                      PRIVATE ${CMAKE_DL_LIBS} ${RESOLV_LIBRARIES}
                              ${CMAKE_THREAD_LIBS_INIT}
                              ${ZLIB_LIBRARIES}
                              ${BZIP2_LIBRARIES}
