diff -w -Nrua a/src/global.c b/src/global.c
--- a/src/global.c	2016-10-17 19:17:54.052797693 +0300
+++ b/src/global.c	2016-10-17 19:26:26.836670681 +0300
@@ -225,22 +225,22 @@
 	TlsSetValue(_tls_index, NULL);
 }
 
-BOOL WINAPI DllMain(HINSTANCE hInstDll, DWORD fdwReason, LPVOID lpvReserved)
-{
-	GIT_UNUSED(hInstDll);
-	GIT_UNUSED(lpvReserved);
+/* BOOL WINAPI DllMain(HINSTANCE hInstDll, DWORD fdwReason, LPVOID lpvReserved) */
+/* { */
+/* 	GIT_UNUSED(hInstDll); */
+/* 	GIT_UNUSED(lpvReserved); */
 
-	/* This is how Windows lets us know our thread is being shut down */
-	if (fdwReason == DLL_THREAD_DETACH) {
-		git__free_tls_data();
-	}
+/* 	/\* This is how Windows lets us know our thread is being shut down *\/ */
+/* 	if (fdwReason == DLL_THREAD_DETACH) { */
+/* 		git__free_tls_data(); */
+/* 	} */
 
-	/*
-	 * Windows pays attention to this during library loading. We don't do anything
-	 * so we trivially succeed.
-	 */
-	return TRUE;
-}
+/* 	/\* */
+/* 	 * Windows pays attention to this during library loading. We don't do anything */
+/* 	 * so we trivially succeed. */
+/* 	 *\/ */
+/* 	return TRUE; */
+/* } */
 
 #elif defined(GIT_THREADS) && defined(_POSIX_THREADS)
 
