diff -Naurw a/source/i18n/number_decimalquantity.cpp b/source/i18n/number_decimalquantity.cpp
--- a/source/i18n/number_decimalquantity.cpp	2019-04-17 22:42:15.000000000 +0300
+++ b/source/i18n/number_decimalquantity.cpp	2019-04-26 18:43:47.132552625 +0300
@@ -446,7 +446,11 @@
         for (; i <= -22; i += 22) n /= 1e22;
         n /= DOUBLE_MULTIPLIERS[-i];
     }
+#ifdef __APPLE__
+    auto result = static_cast<int64_t>(::round(n));
+#else
     auto result = static_cast<int64_t>(std::round(n));
+#endif
     if (result != 0) {
         _setToLong(result);
         scale -= fracLength;
