diff -Naurw a/source/i18n/decimfmt.cpp b/source/i18n/decimfmt.cpp
--- a/source/i18n/decimfmt.cpp	2019-04-17 22:42:15.000000000 +0300
+++ b/source/i18n/decimfmt.cpp	2019-04-26 18:46:32.699690649 +0300
@@ -1812,7 +1812,11 @@
         return false;
     }
     if (std::isnan(input)
+#ifdef __APPLE__
+            || trunc(input) != input
+#else
             || std::trunc(input) != input
+#endif
             || input <= INT32_MIN
             || input > INT32_MAX) {
         return false;
