diff -Naur a/configure b/configure
--- a/configure	2018-05-02 22:35:26.000000000 +0300
+++ b/configure	2018-05-17 22:03:50.467432972 +0300
@@ -5,7 +5,7 @@
 # Report bugs to <https://bugs.gnupg.org>.
 #
 #
-# Copyright (C) 1992-1996, 1998-2012 Free Software Foundation, Inc.
+# Copyright (C) 1992-1996, 1998-2012, 2018 Free Software Foundation, Inc.
 #
 #
 # This configure script is free software; the Free Software Foundation
@@ -8272,7 +8272,11 @@
 # libgpg-error is a library with error codes shared between GnuPG
 # related projects.
 #
-
+if test -n "$GPG_ERROR_VERSION"; then
+    $as_echo "GPG_ERROR_CFLAGS: $GPG_ERROR_CFLAGS"
+    $as_echo "GPG_ERROR_LIBS:   $GPG_ERROR_LIBS"
+    have_gpg_error=yes
+else
   gpg_error_config_prefix=""
 
 # Check whether --with-libgpg-error-prefix was given.
@@ -8414,7 +8418,7 @@
     have_gpg_error=no
   fi
 
-
+fi
 
 
 
@@ -8424,6 +8428,11 @@
 # Libgcrypt is our generic crypto library
 #
 
+if test -n "$LIBGCRYPT_VERSION"; then
+    $as_echo "LIBGCRYPT_CFLAGS: $LIBGCRYPT_CFLAGS"
+    $as_echo "LIBGCRYPT_LIBS:   $LIBGCRYPT_LIBS"
+    have_libgcrypt=yes
+else
 
 # Check whether --with-libgcrypt-prefix was given.
 if test "${with_libgcrypt_prefix+set}" = set; then :
@@ -8593,13 +8602,20 @@
     have_libgcrypt=no
   fi
 
-
+fi
 
 
 
 #
 # libassuan is used for IPC
 #
+if test -n "$LIBASSUAN_VERSION"; then
+    $as_echo "LIBASSUAN_CFLAGS:  $LIBASSUAN_CFLAGS"
+    $as_echo "LIBASSUAN_LIBS:    $LIBASSUAN_LIBS"
+    $as_echo "LIBASSUAN_VERSION: $LIBASSUAN_VERSION"
+    have_libassuan=yes
+    libassuan_version=$LIBASSUAN_VERSION
+else
 
 # Check whether --with-libassuan-prefix was given.
 if test "${with_libassuan_prefix+set}" = set; then :
@@ -8740,7 +8756,7 @@
     LIBASSUAN_LIBS=""
     have_libassuan=no
   fi
-
+fi
 
 
 if test "$have_libassuan" = "yes"; then
@@ -8757,6 +8773,11 @@
 # libksba is our X.509 support library
 #
 
+if test -n "$KSBA_VERSION"; then
+    $as_echo "KSBA_CFLAGS: $KSBA_CFLAGS"
+    $as_echo "KSBA_LIBS:   $KSBA_LIBS"
+    have_ksba=yes
+else
 
 # Check whether --with-ksba-prefix was given.
 if test "${with_ksba_prefix+set}" = set; then :
@@ -8911,7 +8932,7 @@
     KSBA_LIBS=""
     have_ksba=no
   fi
-
+fi
 
 
 
