From d8c38ef6d8057b20b9a6cd288049b2ce50629d61 Mon Sep 17 00:00:00 2001
From: Dmitry Moskalchuk <dm@crystax.net>
Date: Tue, 21 Jun 2016 12:54:50 +0300
Subject: [PATCH 1/6] [android] Fix libcrystax detection

Signed-off-by: Dmitry Moskalchuk <dm@crystax.net>
---
 configure                | 2 ++
 libavformat/os_support.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/configure b/configure
index a4dd55be9f..d0adb15047 100755
--- a/configure
+++ b/configure
@@ -4931,6 +4931,8 @@ EOF
         eval ${pfx}libc_type=klibc
     elif check_${pfx}cpp_condition sys/cdefs.h "defined __BIONIC__"; then
         eval ${pfx}libc_type=bionic
+    elif check_${pfx}cpp_condition sys/cdefs.h "defined __LIBCRYSTAX__"; then
+        eval ${pfx}libc_type=crystax
     elif check_${pfx}cpp_condition sys/brand.h "defined LABELED_BRAND_NAME"; then
         eval ${pfx}libc_type=solaris
         add_${pfx}cppflags -D__EXTENSIONS__ -D_XOPEN_SOURCE=600
diff --git a/libavformat/os_support.h b/libavformat/os_support.h
index caf1a2fab4..48c46ff72c 100644
--- a/libavformat/os_support.h
+++ b/libavformat/os_support.h
@@ -61,11 +61,13 @@
 #  if HAVE_UNISTD_H
 #    include <unistd.h>
 #  endif
+#ifndef __LIBCRYSTAX__
 #  ifdef lseek
 #   undef lseek
 #  endif
 #  define lseek(f,p,w) lseek64((f), (p), (w))
 #endif
+#endif
 
 static inline int is_dos_path(const char *path)
 {
-- 
2.16.1

