diff -Naurw a/libavformat/udp.c b/libavformat/udp.c
--- a/libavformat/udp.c	2018-11-06 02:22:26.000000000 +0300
+++ b/libavformat/udp.c	2019-02-06 22:36:51.072814365 +0300
@@ -1,6 +1,6 @@
 /*
  * UDP prototype streaming system
- * Copyright (c) 2000, 2001, 2002 Fabrice Bellard
+ * Copyright (c) 2000, 2001, 2002, 2019 Fabrice Bellard
  *
  * This file is part of FFmpeg.
  *
@@ -287,12 +287,21 @@
             return AVERROR(EINVAL);
         }
 
+#if defined(__ANDROID__)
+        mreqs.imr_multiaddr = (__u32)((struct sockaddr_in *)addr)->sin_addr.s_addr;
+        if (local_addr)
+            mreqs.imr_interface = (__u32)((struct sockaddr_in *)local_addr)->sin_addr.s_addr;
+        else
+            mreqs.imr_interface = (__u32)INADDR_ANY;
+        mreqs.imr_sourceaddr = (__u32)((struct sockaddr_in *)&sources[i])->sin_addr.s_addr;
+#else
         mreqs.imr_multiaddr.s_addr = ((struct sockaddr_in *)addr)->sin_addr.s_addr;
         if (local_addr)
             mreqs.imr_interface= ((struct sockaddr_in *)local_addr)->sin_addr;
         else
             mreqs.imr_interface.s_addr= INADDR_ANY;
         mreqs.imr_sourceaddr.s_addr = ((struct sockaddr_in *)&sources[i])->sin_addr.s_addr;
+#endif
 
         if (setsockopt(sockfd, IPPROTO_IP,
                        include ? IP_ADD_SOURCE_MEMBERSHIP : IP_BLOCK_SOURCE,
