diff -Naur a/zip.c b/zip.c
--- a/zip.c	2018-02-08 18:22:32.564454091 +0300
+++ b/zip.c	2018-02-08 18:23:13.712708794 +0300
@@ -1,5 +1,5 @@
 /* zip.c -- compress files to the gzip or pkzip format
- * Copyright (C) 1992-1993 Jean-loup Gailly
+ * Copyright (C) 1992-1993, 2018 Jean-loup Gailly
  * This is free software; you can redistribute it and/or modify it under the
  * terms of the GNU General Public License, see the file COPYING.
  */
@@ -11,10 +11,6 @@
 #include <ctype.h>
 #include <sys/types.h>
 
-#include "tailor.h"
-#include "gzip.h"
-#include "crypt.h"
-
 #ifdef HAVE_UNISTD_H
 #  include <unistd.h>
 #endif
@@ -22,6 +18,10 @@
 #  include <fcntl.h>
 #endif
 
+#include "tailor.h"
+#include "gzip.h"
+#include "crypt.h"
+
 local ulg crc;       /* crc on uncompressed file data */
 long header_bytes;   /* number of bytes in gzip header */
 
